src/node_http2.cc

```diff
@@ -811,6 +830,8 @@ int Http2Session::DoWrite(WriteWrap* req_wrap,
     }
   }
 
+  chunks_sent_since_last_write_ = 0;
```

Comment:
Is my understanding correct that this and `SubmitFile` are the *only* parts of the HTTP2 source that can trigger `Http2Session::Send` to be called?

Reply:
As far as I could tell, yes.

Reply:
Sorry, to be clear: it doesn't quite matter if they're the only parts that can trigger it because any trigger of `Send` means that the connection is still active (in some form). The more important part of this is that only this and SubmitFile are triggered when a write occurs and hence the only ones that should reset `chunks_sent_since_last_write_`.

Reply:
Hm, okay. My question wasn’t relevant to this particular PR – it just was kind of my hope that this meant the # of places that could schedule a write would be overseeable so that it would be easy to turn the `uv_prepare_t` we use on/off in those places

Reply:
Right. I think that loop is a bit more complicated since nghttp2 manages the flow of data. The only place that can call `Send` is `SendPendingData` which requests it from `nghttp2_session_mem_send`. I think there are like 2-3 places that call `SendPendingData`.

`Send` itself is responsible for all the data going to the socket, which includes all the frames that nghttp2 itself manages. I think @jasnell might have more insight, I've only started figuring out nghttp2 in the last few weeks.

Reply:
Now that I look at it, SendPendingData is one of those things that is called from the `prep_`... so that wasn't of much help :)

Reply:
Yeah – basically, the question was if these were the only places that would make nghttp2 want to write something to the socket ;) I guess I can answer that for myself with a ”no”, in the end.

Reply:
There are lots of things that could trigger Ng too want to write. Responding to ping frames, for instance, is handled entirely by Ng without any intervention from our code. There are also automatic error handing situations that do not bubble up to our code.

Reply:
> Responding to ping frames, for instance, is handled entirely by Ng without any intervention from our code.

How do we know we need to write to the socket in those cases?

Reply:
That's currently why we attempt to send on every event loop tick. There is a want_write api, however that can be checked. It won't tell you how much data is pending to write, but it will answer true so long as there is data pending in the queue.

Reply:
Thanks, makes sense!